react/features/etherpad/components/native/SharedDocument.js

```diff
@@ -0,0 +1,174 @@
+
+import styles, { INDICATOR_COLOR } from './styles';
+
+/**
+ * The type of the React {@code Component} props of {@code ShareDocument}.
+ */
+type Props = {
+
+    /**
+     * URL for the shared document.
+     */
+    _documentUrl: string,
+
+    /**
+     * Color schemed style of the header component.
+     */
+    _headerStyles: Object,
+
+    /**
+     * True if the chat window should be rendered.
+     */
+    _isOpen: boolean,
+
+    /**
+     * The Redux dispatch function.
+     */
+    dispatch: Dispatch<any>,
+
+    /**
+     * Function to be used to translate i18n labels.
+     */
+    t: Function
+};
+
+/**
+ * Implements a React native component that renders the shared document window.
+ */
+class SharedDocument extends PureComponent<Props> {
+    /**
+     * Instantiates a new instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onClose = this._onClose.bind(this);
+        this._onError = this._onError.bind(this);
+        this._renderLoading = this._renderLoading.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { _documentUrl, _isOpen } = this.props;
+        const webViewStyles = this._getWebViewStyles();
+
+        return (
+            <SlidingView
+                onHide = { this._onClose }
+                position = 'bottom'
+                show = { _isOpen } >
+                <View style = { styles.webViewWrapper }>
+                    <HeaderWithNavigation
+                        headerLabelKey = 'documentSharing.title'
+                        onPressBack = { this._onClose } />
+                    <SafeAreaView style = { webViewStyles }>
+                        <WebView
+                            onError = { this._onError }
+                            renderLoading = { this._renderLoading }
+                            source = {{ uri: _documentUrl }}
+                            startInLoadingState = { true } />
+                    </SafeAreaView>
+                </View>
+            </SlidingView>
+        );
+    }
+
+    /**
+     * Computes the styles required for the WebView component.
+     *
+     * @returns {Object}
+     */
+    _getWebViewStyles() {
+        return {
+            ...styles.webView,
+            backgroundColor: this.props._headerStyles.screenHeader.backgroundColor
+        };
+    }
+
+    _onClose: () => boolean
+
+    /**
+     * Closes the window.
+     *
+     * @returns {boolean}
+     */
+    _onClose() {
+        const { _isOpen, dispatch } = this.props;
+
+        if (_isOpen) {
+            dispatch(toggleDocument());
+
+            return true;
+        }
+
+        return false;
+    }
+
+    _onError: () => void;
+
+    /**
+     * Callback to handle the error if the page fails to load.
+     *
+     * @returns {void}
+     */
+    _onError() {
+        this.props.dispatch(toggleDocument());
```

Comment:
Is it possible an `_isOpen` check is needed here too?

Reply:
Yeah, I’ll add it just in case.